### PR TITLE
ARM Support

### PR DESCRIPTION
**Pull request by aliariff**
[Rendered](https://github.com/aliariff/gsoc/blob/arm-support/rfc/2020/arm-support/aliariff.md)

### CODE REVIEW

**Comment by ihcsim** [comment]
@aliariff Thanks for the RFC! I know there are many in the community who are excited about this.

I assume this will include updating the proxy and proxy-init image too, right?

**Comment by ihcsim** [comment]
@aliariff Please go ahead and update this PR with your design proposal.

**Comment by aliariff** [comment]
@ihcsim Yes, updating proxy-init and proxy are required. Ok, I will proceed.

File: rfc/2020/arm-support/aliariff.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    .
+    .
+    bin/docker-push $TAG
+    DOCKER_BUILD_ARCH=arm64 bin/docker-push $TAG
+    DOCKER_BUILD_ARCH=arm bin/docker-push $TAG
+
+    bin/docker-manifest $TAG
+    ```
+
+With all of step are explain, there is some drawback that we can see here as the we need to create manifest for multi-arch our self and every image architecture will have their own repositories.
```
Comment (ihcsim) [inline rfc/2020/arm-support/aliariff.md]: Last I tried, the `docker manifest` command is still experimental. I don't think there is a way to enable it on GitHub Actions.

> every image architecture will have their own repositories.

Can we assign different tags to the images, but store them in the same repository?

File: rfc/2020/arm-support/aliariff.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    FROM --platform=$GOARCH ...
+    ```
+  - Adjust `bin/fetch-proxy` file to retrieve the correct architecture version
+  - Add `bin/docker-manifest` file to create and push multi-arch manifest
+    ```sh
+    for img in cli-bin cni-plugin controller debug grafana proxy web ; do
+      docker manifest create gcr.io/linkerd-io/$img:$tag \
+        gcr.io/linkerd-io/$img-amd64:$tag \
+        gcr.io/linkerd-io/$img-arm64:$tag \
+        gcr.io/linkerd-io/$img-arm:$tag
```
Comment (ihcsim) [inline rfc/2020/arm-support/aliariff.md]: I assume we need a `docker manifest push` command after this.

File: rfc/2020/arm-support/aliariff.md (hunk 1 above)
Comment (aliariff) [reply rfc/2020/arm-support/aliariff.md]: I just try. It is possible to run experimental features on the Github Action runner.
image
Ref: https://github.com/actions/virtual-environments/issues/368

Comment (aliariff) [reply rfc/2020/arm-support/aliariff.md]: Different tags and the same repository are also possible. But that repo will look messy. But yeah totally possible. 
I create this by following how grafana team created it. Ref: https://hub.docker.com/u/grafana

File: rfc/2020/arm-support/aliariff.md (hunk 2 above)
Comment (aliariff) [reply rfc/2020/arm-support/aliariff.md]: You are right missing the push part. Will add it.

**Review by ihcsim — APPROVED** [review]

**Pull request merged**